pkg/controller/common/reconcile_result.go

```diff
@@ -0,0 +1,19 @@
+package common
+
+import (
+	"time"
+
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func Reconciled() (reconcile.Result, error) {
+	return reconcile.Result{}, nil
+}
+
+func RequeueWithError(err error) (reconcile.Result, error) {
+	return reconcile.Result{}, err
+}
+
+func RequeueWithTimeout(timeout time.Duration) (reconcile.Result, error) {
+	return reconcile.Result{RequeueAfter: timeout}, nil
+}
```

Comment:
Inspired by: https://cloud.redhat.com/blog/7-best-practices-for-writing-kubernetes-operators-an-sre-perspective

I didn't change existing code which returns `reconcile.Result{}` directly, because the PR would be huge and I didn't want to increase its scope.
If you don't like this, because of inconsistent code style I can revert it. I just thought that we can replace old code gradually along other changes.

Reply:
No, this is a great improvement. Thanks!